src/category_theory/action.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+
+import category_theory.groupoid
+import category_theory.endomorphism
+import category_theory.elements
+import group_theory.group_action
+import category_theory.single_obj
+
+/-!
+# Actions as functors and as categories
+
+From a multiplicative action M ↻ X, we can construct a functor from M to the category of
+types, mapping the single object of M to X and an element `m : M` to map `X → X` given by
+multiplication by `m`.
+  This functor induces a category structure on X -- a special case of the category of elements.
+A morphism `x → y` in this category is simply a scalar `m : M` such that `m • x = y`. In the case
+where M is a group, this category is a groupoid -- the `action groupoid'.
+-/
+
+open mul_action
+namespace category_theory
+
+universes u
+
+variables (M : Type*) [monoid M] (X : Type u) [𝒜 : mul_action M X]
+include 𝒜
+
+/-- A multiplicative action M ↻ X viewed as a functor in the natural way. -/
+@[reducible]
+def action_as_functor : single_obj M ⥤ Type u :=
+{ obj := λ _, X,
+  map := λ _ _, (•),
+  map_id' := λ _, funext $ mul_action.one_smul M,
+  map_comp' := λ _ _ _ f g, funext $ λ x, (smul_smul g f x).symm }
+
+/-- A multiplicative action M ↻ X induces a category strucure on X, where a morphism
+ from x to y is a scalar taking x to y. Due to implementation details, the object type
+ of this category is not equal to X, but is in bijection with X. -/
+def action_category := (action_as_functor M X).elements
+
+namespace action_category
+
+instance : category (action_category M X) :=
+category_theory.category_of_elements _
+
+example (p q : action_category M X) : (p ⟶ q) = { m : M // m • p.2 = q.2 } := rfl
```

Comment:
Why not make this a lemma?

Reply:
I wasn't sure how to treat this fact, but it probably should be a lemma.